src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -114,19 +114,19 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
         return [];
     }
 
-    private function mapQueryString(Request $request, string $type, MapQueryString $attribute): ?object
+    private function mapQueryString(Request $request, ArgumentMetadata $argument, MapQueryString $attribute): ?object
     {
         if (!$data = $request->query->all()) {
             return null;
         }
 
-        return $this->serializer->denormalize($data, $type, null, self::CONTEXT_DENORMALIZE + $attribute->context);
+        return $this->serializer->denormalize($data, $argument->getType(), null, self::CONTEXT_DENORMALIZE + $attribute->context);
     }
 
-    private function mapRequestPayload(Request $request, string $type, MapRequestPayload $attribute): ?object
+    private function mapRequestPayload(Request $request, ArgumentMetadata $argument, MapRequestPayload $attribute): ?object
     {
         if ($data = $request->request->all()) {
```

Comment:
we should check the format before this if, so that forms are not accepted if only format "json" is listed for example

Reply:
changed, thanks